File: Linked List/IntersectionOfTwoLL.java
// https://leetcode.com/problems/intersection-of-two-linked-lists/
public class Solution {
    public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
        ListNode a = headA, b = headB;
        while (a != b) {
            a = a.next;
            b = b.next;
            if (a == null && b == null) break;
            if (a == null) a = headB;
            if (b == null) b = headA;
        }
        return a;
    }
}

File: Linked List/IntersectionOfTwoLL.cpp
// https://leetcode.com/problems/intersection-of-two-linked-lists/
class Solution {
public:
    ListNode *getIntersectionNode(ListNode *headA, ListNode *headB) {
        ListNode *a = headA, *b = headB;
        while (a != b) {
            a = a->next;
            b = b->next;
            if (a == NULL && b == NULL) break;
            if (a == NULL) a = headB;
            if (b == NULL) b = headA;
        }
        return a;
    }
};
